src/main/java/gregtech/api/recipes/Recipe.java

```diff
@@ -198,14 +193,10 @@ public boolean matches(boolean consumeIfSuccessful, List<ItemStack> inputs, List
 
         for (FluidStack fluid : this.fluidInputs) {
             int fluidAmount = fluid.amount;
-            boolean isNotConsumed = false;
-            if (fluidAmount == 0) {
-                fluidAmount = 1;
-                isNotConsumed = true;
-            }
+            boolean isNotConsumed = (fluid.tag != null && fluid.tag.hasKey("nonConsumable"));
             for (int i = 0; i < fluidInputs.size(); i++) {
                 FluidStack tankFluid = fluidInputs.get(i);
-                if (tankFluid == null || !tankFluid.isFluidEqual(fluid))
+                if (tankFluid == null || tankFluid.getFluid() != fluid.getFluid())
                     continue;
                 int fluidAmountToConsume = Math.min(fluidAmountInTank[i], fluidAmount);
                 fluidAmount -= fluidAmountToConsume;
```

Comment:
Same question here as in the item section.

Reply:
the notconsumed boolean is determined by the existence of NBT tag "nonConsumable"